Process_Library/GOCART2G_MieMod.F90

```diff
@@ -537,26 +556,35 @@ integer function getChannel(this, wavelength, rc) result (ch)
 
   end function getChannel
 
-  real function getWavelength(this, ith_channel, rc) result (wavelength)
+  real function getWavelength(this, channel, rc) result (wavelength)
      class (GOCART2G_Mie), intent(in) :: this
-     integer, intent(in) :: ith_channel
+     integer, intent(in) :: channel
      integer, optional, intent(out) :: rc
      real, parameter :: w_tol = 1.e-9
      integer :: i
 
      if (present(rc)) rc = 0
 
-     if (ith_channel <=0 .or. ith_channel > this%nch ) then
+     if (channel <=0 .or. channel > this%nch ) then
        !$omp critical
-       print*, "The channel of ",ith_channel, " is an invalid channel number."
+       print*, "The channel of ",channel, " is an invalid channel number."
        !$omp end critical
        if (present(rc)) rc = -1
        wavelength = -1. ! meanlingless nagative
        return
      endif
 
-     wavelength = this%wavelengths(ith_channel)
+     wavelength = this%wavelengths(channel)
 
   end function getWavelength
 
+  elemental function interp(this,table, irh, arh)
+    class (GOCART2G_Mie), intent(in) :: this
+    type(RH_Table), intent(in) :: table
+    integer, intent(in) :: irh
+    real, intent(in) :: arh
+    real :: interp
+    interp = sum(table%rh(irh:irh+1) * [(1-arh),arh])
+  end function interp
+
```

Comment:
This should not be a type-bound procedure.   It does not use `this` inside.

Reply:
Done